src/tests/ftest/security/pool_connect_init.py

```diff
@@ -0,0 +1,146 @@
+  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+
+  GOVERNMENT LICENSE RIGHTS-OPEN SOURCE SOFTWARE
+  The Government's rights to use, modify, reproduce, release, perform, display,
+  or disclose this software are subject to the terms of the Apache License as
+  provided in Contract No. B609815.
+  Any reproduction of computer software, computer software documentation, or
+  portions thereof marked with this legend must also reproduce the markings.
+'''
+from __future__ import print_function
+import os
+import traceback
+from apricot import TestWithServers
+from daos_api import DaosPool, DaosApiError
+
+class PoolSecurityTest(TestWithServers):
+    """
+    DAOS security: initial security tests after pool creation
+    :avocado: recursive
+    """
+    def setUp(self):
+        """
+        set up method: create 3 pools with different uid and gid
+                       for security test.
+            pool1: with owner uid, owner-group root-id.
+            pool2: with owner rooter-id, owner-group gid.
+            pool3: owner and owner-group with both root-id.
+        """
+        super(PoolSecurityTest, self).setUp()
+        # get parameters from yaml file
+        mode = self.params.get("mode", '/run/poolparams/createmode/', 511)
+        setid = self.params.get("setname", '/run/poolparams/createset/',
+                                "daos_server")
+        size = self.params.get("size", '/run/poolparams/createsize/',
+                               107374182)
+        root_id = 0
+        user_uid = os.geteuid()
+        user_gid = os.getegid()
+        try:
+            self.log.info("===Create DAOS pool1 with:")
+            self.log.info("===uid:  %s", user_uid)
+            self.log.info("===gid:  %s", root_id)
+            self.pool1 = DaosPool(self.context)
+            self.pool1.create(mode, user_uid, root_id, size, setid)
+            self.log.info("===>Pool1 create succeed.")
+            self.log.info("===Create DAOS pool2 with:")
+            self.log.info("===uid:  %s", root_id)
+            self.log.info("===gid:  %s", user_gid)
+            self.pool2 = DaosPool(self.context)
+            self.pool2.create(mode, root_id, user_gid, size, setid)
+            self.log.info("===>Pool2 create succeed.")
+            self.log.info("===Create DAOS pool3 with:")
+            self.log.info("===uid:  %s", root_id)
+            self.log.info("===gid:  %s", root_id)
+            self.pool3 = DaosPool(self.context)
+            self.pool3.create(mode, root_id, root_id, size, setid)
+            self.log.info("===>Pool3 create succeed.")
```

Comment:
It seems like each creation sequence and outputs are basically the same for each pool, except a few things that could be passed as parameters. Would it make sense to break out a helper function that accepts name, owner, and owner-group as params?

Reply:
Updated. please see the new commit.